pkg/mdformatter/linktransformer/link.go

```diff
@@ -239,6 +251,15 @@ func NewValidator(ctx context.Context, logger log.Logger, linksValidateConfig []
 	if config.Timeout != "" {
 		v.c.SetRequestTimeout(config.timeout)
 	}
+
+	if v.validateConfig.CacheType == sqlite && storage != nil {
+		v.storage = storage
+		v.storage.Validity = v.validateConfig.CacheValidity
```

Comment:
This looks like a code smell.

We pass storage here for some composability, but we kind of  use it like struct (setting Field directly) despite doing  `Init`  below. Can we add `Validity` parameter to `Init` then?

Reply:
Will try to restructure this!